sphinx/domains/javascript.py

```diff
@@ -106,20 +106,19 @@ def add_target_and_index(self, name_obj: Tuple[str, str], sig: str,
                              signode: desc_signature) -> None:
         mod_name = self.env.ref_context.get('js:module')
         fullname = (mod_name + '.' if mod_name else '') + name_obj[0]
-        if fullname not in self.state.document.ids:
-            signode['names'].append(fullname)
-            signode['ids'].append(fullname.replace('$', '_S_'))
```

Comment:
It might be better to generate an old styled anchor as an alias for a while. It will helps to old hyperlinks. But that also means we will not remove old styled ones in future... Is there any opinion?

Reply:
I think no reason to not to support old styled anchors.

Reply:
done